src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -552,12 +552,16 @@ def dbs_get_all(dbs, db_name, _hash, *args, **kwargs):
         db get_all function executed on global and all namespace DBs.
         """
         result = {}
+        Namespace.connect_all_dbs(dbs, db_name)
+        if len(dbs) == 1:
+            return dbs[0].get_all(db_name, _hash, *args, **kwargs)
         for db_conn in dbs:
-            db_conn.connect(db_name)
-            if(db_conn.exists(db_name, _hash)):
-                ns_result = db_conn.get_all(db_name, _hash, *args, **kwargs)
-                if ns_result is not None:
-                    result.update(ns_result)
+            # If there are multiple namespaces, _hash might not be 
+            # present in all namespace, ignore if not present in a
+            # specfic namespace.
+            ns_result = db_conn.get_all(db_name, _hash, blocking=False)
```

Comment:
we should same parameters for get_all for single vs multi-asic call . 
we can update kwargs['blocking'] if present to False if len(dbs) > 1 (Muti-asic scenario)

Reply:
updated as per comment.